pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -330,12 +333,17 @@ def __call__(
             # Calculate the length of the work arrays
             # --------------------------------------
             nvar = self.optProb.ndvs
-            lencw = 500
-            leniw = 500 + 100 * (ncon + nvar)
-            lenrw = 500 + 200 * (ncon + nvar)
-
-            self.setOption("Total integer workspace", leniw)
-            self.setOption("Total real workspace", lenrw)
+            lencw = self.getOption("Total character workspace")
+            leniw = self.getOption("Total integer workspace")
+            lenrw = self.getOption("Total real workspace")
+
+            # Set defaults
+            if leniw is None:
+                leniw = 500 + 100 * (ncon + nvar)
```

Comment:
Can we define this magic number 500 anywhere, or better, connect it with and replace it with `lencw` in case that gets updated from the interface. Same below.

Reply:
500 is the minimum workspace length. SNOPT will exit with `81` if any of the lengths are lower than that. I guess I could define a variable for that? It's not really connected to `lencw`.

Reply:
Right, exactly its the minimum for all the work arrays. My comment was more to make it more explicit that this number is the minimum in the SNOPT code itself, i.e. remind the reader of the python code why its 500, not just a random number. I suggest we at least add a comment on what this number is. Alternatively, we can also define a variable `minWorkArraySize = 500` or something.  
And you are right its not connected and they dont have to be connected. `lencw` is the only variable out of the three in the options with a default value set. I imagined that in case you change this number its likely that you need to modify the others too, so by making this connected you would essentially only need to change one instead of all three. 
However, its probably best *not* to make them connected as it better reflects the source code, and if you are going to change these numbers yourself anyways you probably have a good reason to do so and will likely need to modify all three.